src/Symfony/Component/Form/Extension/Core/EventListener/ResizeFormListener.php

```diff
@@ -44,15 +48,50 @@ public function __construct(
     public static function getSubscribedEvents(): array
     {
         return [
-            FormEvents::PRE_SET_DATA => 'preSetData',
+            FormEvents::PRE_SET_DATA => 'preSetData', // deprecated
+            FormEvents::POST_SET_DATA => ['postSetData', 255], // as early as possible
             FormEvents::PRE_SUBMIT => 'preSubmit',
             // (MergeCollectionListener, MergeDoctrineCollectionListener)
             FormEvents::SUBMIT => ['onSubmit', 50],
         ];
     }
 
+    /**
+     * @deprecated Since Symfony 7.2, use {@see postSetData()} instead.
+     */
     public function preSetData(FormEvent $event): void
     {
+        if (__CLASS__ === static::class
+            || __CLASS__ === (new \ReflectionClass($this))->getMethod('preSetData')->getDeclaringClass()->name
+        ) {
+            // not a child class, or child class does not overload PRE_SET_DATA
+            return;
+        }
+
+        trigger_deprecation('symfony/form', '7.2', 'Calling "%s()" is deprecated, use "%s::postSetData()" instead.', __METHOD__, __CLASS__);
+        // parent::preSetData() has been called
+        $this->overridden = false;
+        $this->postSetData($event);
+        $this->usePreSetData = true;
+    }
+
+    // Remove FormEvent type hint in 8.0
+    final public function postSetData(FormEvent|PostSetDataEvent $event): void
```

Comment:
`@final since Symfony 7.3` insead
```suggestion
    /**
     * @param PostSetDataEvent $event
     */
    public function postSetData(FormEvent|PostSetDataEvent $event): void
```